go-controller/pkg/node/default_node_network_controller.go

```diff
@@ -1440,6 +1464,156 @@ func (nc *DefaultNodeNetworkController) WatchNamespaces() error {
 	return err
 }
 
+func (nc *DefaultNodeNetworkController) WatchNodes() error {
+	_, err := nc.retryNodes.WatchResource()
+	return err
+}
+
+// addOrUpdateNode handles creating routes or nftables rules for each node to handle PMTUD
+func (nc *DefaultNodeNetworkController) addOrUpdateNode(node *corev1.Node) error {
+	hostCIDRs := nc.Gateway.GetHostCIDRs()
+	var nftElems []*knftables.Element
+	for _, address := range node.Status.Addresses {
+		if address.Type != corev1.NodeInternalIP {
+			continue
+		}
+		nodeIP := net.ParseIP(address.Address)
+		if nodeIP == nil {
+			continue
+		}
+		var existingHostSubnet *net.IPNet
+		for _, hostCIDR := range hostCIDRs {
+			if utilnet.IsIPv6CIDR(hostCIDR) != utilnet.IsIPv6(nodeIP) {
+				continue
+			}
+			if hostCIDR.Contains(nodeIP) {
+				existingHostSubnet = hostCIDR
+				break
+			}
+		}
+
+		// subnet exists on our node
+		if existingHostSubnet != nil {
+			// create MTU lock routes
+			mask := 32
+			nlFamily := netlink.FAMILY_V4
+			if utilnet.IsIPv6(nodeIP) {
+				mask = 128
+				nlFamily = netlink.FAMILY_V6
+			}
+			routes, err := util.GetNetLinkOps().RouteList(nil, nlFamily)
+			if err != nil {
+				return fmt.Errorf("unable to list host routes: %w", err)
+			}
+			linkIndex := -1
+			existingRoute := netlink.Route{}
+			for _, route := range routes {
+				if existingHostSubnet.Contains(route.Dst.IP) {
+					linkIndex = route.LinkIndex
+					existingRoute = route
+					break
+				}
```

Comment:
can you stop programming the route if an existing route exists that matches what we want?

Reply:
hmm i guess i could use routemanager equal method to see if they are the same and then skip processing it if so. That compares mtu lock as well. I'll look into it.